06_puppy/runnerdave/storer_test.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	puppy1 = func() Puppy {
+		return Puppy{ID: 11, Breed: "Chihuahua", Colour: "Brown", Value: 12.30}
+	}
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+}
+
+func (s *storerSuite) TestUpdatePuppyIDDoesNotExist() {
+	// given
+	assert := tassert.New(s.T())
+	testPuppy := puppy1()
+
+	// when
+	err := s.store.UpdatePuppy(11, &testPuppy)
+
+	// then
+	assert.NoError(err, "Updating to non-existing puppy ID is not an error")
+	newPuppy, err := s.store.ReadPuppy(11)
+	if assert.NoError(err, "Should be able to read a newly added puppy via update") {
+		assert.Equal(&testPuppy, newPuppy, "Newly added puppy should be equal to test puppy")
+	}
+}
+
+func (s *storerSuite) TestReadPuppy() {
+	// given
+	assert := tassert.New(s.T())
+	testPuppy := puppy1()
+	createError := s.store.CreatePuppy(&testPuppy)
+	s.T().Log(createError)
+
+	// when
+	newPuppy, err := s.store.ReadPuppy(11)
+
+	// then
+	if assert.NoError(err, "Should be able to read a newly added puppy") {
+		assert.Equal(&testPuppy, newPuppy, "Newly added puppy should be equal to test puppy")
+	}
+}
+
+func (s *storerSuite) TestDeletePuppy() {
+	// given
+	assert := tassert.New(s.T())
+	testPuppy := puppy1()
+	createError := s.store.CreatePuppy(&testPuppy)
+	s.T().Log(createError)
```

Comment:
you can use testify require to terminate the test here if you get an error on create
```
r := require.New(t)
r.NoError(err, "Create should not produce an error")
```

Reply:
implemented